autoload/neomake/makers/ft/c.vim

```diff
@@ -22,6 +24,26 @@ function! neomake#makers#ft#c#clang()
         \ }
 endfunction
 
+function! neomake#makers#ft#c#clangcheck()
+    if !exists('g:neomake_c_clangcheck_config_file')
+        let g:neomake_c_clangcheck_config_file = '.neomake_clangcheck_config'
+    endif
+
+    return {
+        \ 'exe': 'clang-check',
+        \ 'args': ['%:p', '--', '-fdiagnostic-format=clang']
+                  \ + neomake#utils#ReadConfig(g:neomake_c_clangcheck_config_file),
```

Comment:
Issue with that is that different projects need different flags, and having a config file in the project is the only solution to that that feels sane to me.